TOY-61 커스텀 갤러리화면 Kotlin 변환

pby2017 opened the pull request:
코드 리뷰 부탁드립니다.

이후 TOY-69에서 MVVM 아키텍처 적용 작업을 추가로 할 계획입니다.

pby2017 committed:
TOY-61 커스텀 갤러리화면 Kotlin 변환

이후 TOY-69에서 MVVM 아키텍처 적용 작업을 추가로 할 계획입니다.

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/gallery/GalleryFragment.kt:
```diff
@@ -0,0 +1,95 @@
+class GalleryFragment : Fragment() {
+
+    private val galleryAdapter by lazy { GalleryAdapter(context, ArrayList()) }
+    private val gridLayoutManager by lazy { GridLayoutManager(context, 3) }
```
굳이 인스턴스 변수로 써야하나 싶네요.

pby2017 replied:
필요한 곳이 한 곳 뿐이어서 인스턴스 변수로 써야할 필요가 없었습니다.

피드백 반영해서 변수로 만들지 않고
필요한 곳에서 바로 인스턴스화 하도록 수정하겠습니다.

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/gallery/GalleryFragment.kt:
```diff
@@ -0,0 +1,95 @@
+
+        fun loadImages(folderName: String?): List<LoadImage> {
+            val listOfAllImages = ArrayList<LoadImage>()
+            GlobalApplication.globalApplicationContext.contentResolver.query(MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
```
contentResolver에 query하는 코드가 중복되는거 같은데 따로 util로 빼는게 좋겠네요

pby2017 replied:
네,
피드백 반영해서 query 코드를 util 파일로 빼겠습니다.

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/gallery/GalleryAdapter.kt:
```diff
@@ -0,0 +1,85 @@
+
+import kr.co.yogiyo.rookiephotoapp.R
+
+class GalleryAdapter(private val context: Context?, private var loadImages: List<LoadImage>) : RecyclerView.Adapter<GalleryAdapter.GalleryViewHolder>() {
```
Adapter도 viewModel을 갖는 형태로 시도해보는 것도 좋겠어요.

pby2017 replied:
네,
TODO 작성해놓고
MVVM 적용하면서 피드백 반영하겠습니다.

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/gallery/GalleryAdapter.kt:
```diff
@@ -0,0 +1,85 @@
+
+    private var selectedViewHolder: GalleryViewHolder? = null
+
+    val selectedImageUri: Uri?
```
selectedViewHolder가 null인 상황이 절대 없는건가요?

pby2017 replied:
null인 상황이 있습니다.
피드백 반영해서 null 처리 연산자를 수정하겠습니다. (`!!` -> `?.`)

bonchangkoo reviewed (commented):
코멘트 확인 부탁해요.

hyejin830 commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/gallery/GalleryActivity.kt:
```diff
@@ -0,0 +1,196 @@
+            close()
+        } ?: return ArrayList()
+
+        return ArrayList<String>().apply {
```
[질문입니다] 위에 80line의 return에서 작성해도 될 것 같은데 이렇게 나누신 이유가 있으실까요?!

hyejin830 commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/gallery/GalleryActivity.kt:
```diff
@@ -0,0 +1,196 @@
+        if (resultCode == Activity.RESULT_OK) {
+            if (requestCode == Constants.REQUEST_DIARY_PICK_GALLERY) {
+                Intent(this@GalleryActivity, DiaryEditActivity::class.java)
+                        .setData(data?.data).let {
```
`data?.data` 이 부분이 가독성이 좀 떨어지는 것 같아요
다이어리 편집 페이지로 인텐트할 때, uri 데이터가 있어야 하니까 아예 intent data가 널이 아닐 경우만 처리하는 게 더 좋을 것 같아요!

```
data?.run {
            if (resultCode == Activity.RESULT_OK) {
                if (requestCode == Constants.REQUEST_DIARY_PICK_GALLERY) {
                    Intent(this@GalleryActivity, DiaryEditActivity::class.java)
                            .setData(this.data).let {
                                setResult(Activity.RESULT_OK, it)
                            }
                    finish()
                }
            }
}
```

pby2017 replied:
네,
피드백 반영해서

함수의 파라미터로 넘어오는 data 이름을 intent로 수정하고, (data -> intent)
제안주신 코드로 반영하겠습니다.

hyejin830 commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/gallery/GalleryActivity.kt:
```diff
@@ -0,0 +1,196 @@
+        val mapOfAllImageFolders = HashMap<String, Int>()
+        var countOfAllImages = 0
+
+        contentResolver.query(MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
```
제 개인적인 의견인데.. run, let 람다를 많이 쓰셔서 코드 읽기가 많이 어렵네요..

pby2017 replied:
네,
피드백 반영해서 수정했습니다.

pby2017 committed:
TOY-61 피드백 반영

pby2017 committed:
TOY-61 충돌 해결 및 비정상 코드 수정

Merge branch 'develop' into TOY-61-gallery-custom-kotlin

# Conflicts:
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/gallery/GalleryActivity.java

bonchangkoo reviewed (commented):
확인했습니다.

hyejin830 reviewed (approved):
확인 완료했습니다
